openassessment/xblock/studio_mixin.py

```diff
@@ -169,20 +252,10 @@ def update_editor_context(self, data, suffix=''):
             logger.exception('Editor context is invalid')
             return {'success': False, 'msg': self._('Error updating XBlock configuration')}
 
-        # Check that the editor assessment order contains all the assessments.  We are more flexible on example-based.
-        given_without_example_based = set(data['editor_assessments_order']) - {'example-based-assessment'}
-        if set(DEFAULT_EDITOR_ASSESSMENTS_ORDER) != given_without_example_based:
-            # Backwards compatibility: "staff-assessment" may not be present.
-            # If that is the only problem with this data, just add it manually and continue.
-            if set(DEFAULT_EDITOR_ASSESSMENTS_ORDER) == (
-                # Check the given set, minus example-based, plus staff
-                given_without_example_based | {'staff-assessment'}
-            ):
-                data['editor_assessments_order'].append('staff-assessment')
-                logger.info('Backwards compatibility: editor_assessments_order now contains staff-assessment')
-            else:
-                logger.exception('editor_assessments_order does not contain all expected assessment types')
-                return {'success': False, 'msg': self._('Error updating XBlock configuration')}
```

Comment:
This block is now part of the validation logic, see `# Include any additional assessments that may have been added since the problem was created.` above